crates/api_crud/src/comment/create.rs

```diff
@@ -67,31 +67,35 @@ impl PerformCrud for CreateComment {
       return Err(LemmyError::from_message("locked"));
     }
 
-    // If there's a parent_id, check to make sure that comment is in that post
-    if let Some(parent_id) = data.parent_id {
-      // Make sure the parent comment exists
-      let parent = blocking(context.pool(), move |conn| Comment::read(conn, parent_id))
+    // Fetch the parent, if it exists
+    let parent_opt = if let Some(parent_id) = data.parent_id {
+      blocking(context.pool(), move |conn| Comment::read(conn, parent_id))
         .await?
-        .map_err(|e| LemmyError::from_error_message(e, "couldnt_create_comment"))?;
+        .ok()
+    } else {
+      None
+    };
 
-      // Strange issue where sometimes the post ID is incorrect
+    // If there's a parent_id, check to make sure that comment is in that post
+    // Strange issue where sometimes the post ID of the parent comment is incorrect
```

Comment:
Should try to fix this before release. Maybe open an issue about it.

Reply:
Someone coded a bug in their client one time that put the post_id in the parent_comment_id field, and it had some weird results. We have to reject requests that provide faulty data.